sktime/annotation/hmm.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+"""
+HMM Annotation Estimator.
+
+Implements a basic Hidden Markov Model (HMM) as an annotation estimator.
+To read more about the algorithm, check out the wikipedia page:
+(https://en.wikipedia.org/wiki/Hidden_Markov_model
+"""
+from math import isclose
+from typing import Tuple
+
+import numpy as np
+
+from sktime.annotation.base._base import BaseSeriesAnnotator
+
+__author__ = ["miraep8"]
+__all__ = ["HMM"]
+
+
+class HMM(BaseSeriesAnnotator):
+    """Implements a simple HMM fitted with viterbi algorithm.
+
+    The HMM annotation estimator facilitates the appilcation of
+    the vertibi algorithm to fit a sequence of 'hidden state' class
+    annotations (represented by an array of integers the same size
+    as the observation) to a sequence of observations.
+
+    This is done by finding the most likely path given the emission
```

Comment:
It would be crucial to be more precise to describe what the output is, given input.
I.e., given the X, what is being produced in `predict`? What is the fit doing?

The current docstring is just a high-level description of HMM, not of the specific procedure that fit/predict below carries out.

Reply:
I added some more details to that effect (and also moved all the logic to predict as you had suggested). Are you happier with it now, or does it still need more?